src/preamp.jl

```diff
@@ -97,30 +125,38 @@ RDWaveform, GenericPreAmp -> RDWaveform
 Simulate the effecs of the preamp on the waveform.
 """
 function simulate(wf::RDWaveform, preamp::GenericPreAmp)
+
+    ## noise before preamp
+    # Alessandro Razeto said noise should be added before preamp
+    # if I simply move adding Gaus noise here, it gives weird cyclical vibrations in the waveform
+    # then after build_dsp there is no Amax at all, I guess DSP fails
+    # wf_noise = simulate_noise(wf, preamp)
+
     ## rise and decay time
     csa_filter = RadiationDetectorDSP.simple_csa_response_filter(
         preamp.τ_rise / step(wf.time),
         uconvert(u"ns", preamp.τ_decay) / step(wf.time))
 
-    wf_preamp = RDWaveform(wf.time, filt(csa_filter, wf.signal))
+    wf_preamp = RDWaveform(wf.time, filt(csa_filter, wf.signal)) 
 
-    ## noise
-    wf_preamp = simulate_noise(wf_preamp, preamp) 
+    wf_preamp = simulate_noise(wf_preamp, preamp)
 
     ## offset
     # wf values are in eV but without u"eV" attached
-    offset = ustrip(uconvert(u"eV", preamp.offset))
-    wf_preamp = RDWaveform(wf_preamp.time, wf_preamp.signal .+ offset)
+    offset_keV = ustrip(uconvert(u"eV", preamp.offset_keV))
```

Comment:
The naming is a bit weird because `offset_keV` now has units of `eV`.

Reply:
You're right, fixed it.
Actually ToDo to untangle the unit threads here and see why waveform signal is not `u"eV"` but "unitless eV". Would be better practice to keep the unit.

Reply:
I agree.